server/src/main/java/org/opensearch/snapshots/ConcurrentSnapshotExecutionException.java

```diff
@@ -58,6 +58,6 @@ public ConcurrentSnapshotExecutionException(StreamInput in) throws IOException {
 
     @Override
     public RestStatus status() {
-        return RestStatus.SERVICE_UNAVAILABLE;
+        return RestStatus.TOO_MANY_REQUESTS;
     }
```

Comment:
Thanks can you please also add a test?

Reply:
I am wondering shouldn't this be 409 (Conflict) instead 429?

Reply:
What's the underlying error? Is it concurrent snapshots being taken whereas this isn't supported? In that case it should definitely not be a 429, that implies the server is unable to handle traffic, not a predictable conflicting behavior - 409 seems more correct in that case.

Reply:
@baba-devv : Since it is concurrent, it is resulting in conflicting and hence rejected by server.

Reply:
@baba-devv yes in my opinion it should be 409